hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemTransactions.java

```diff
@@ -377,95 +365,110 @@ ctx, createFileID(filesConfig.hapiPermissions(), config), bytes),
         if (autoNodeAdminKeyUpdates.tryIfPresent(adminConfig.upgradeSysFilesLoc(), systemContext)) {
             dispatch.stack().commitFullStack();
         }
+        // Dispatch node updates for all the nodes whose node account's account number is less
+        // than 100, with decline_reward set to true
+        final var ledgerConfig = config.getConfigData(LedgerConfig.class);
+        final var nodeStore = dispatch.handleContext().storeFactory().readableStore(ReadableNodeStore.class);
+        for (int i = 0; i < nodeStore.sizeOfState(); i++) {
+            final var node = nodeStore.get(i);
+            final var nodeInfo = networkInfo.nodeInfo(i);
+            if (nodeInfo != null && node != null && !node.deleted()) {
+                final var declineReward = nodeInfo.accountId().accountNumOrThrow() <= ledgerConfig.numSystemAccounts();
+                log.info(
+                        "Dispatching node update for node {} with node account id {} decline_reward set to {}",
+                        nodeInfo.nodeId(),
+                        nodeInfo.accountId(),
+                        declineReward);
+                systemContext.dispatchAdmin(b -> b.nodeUpdate(NodeUpdateTransactionBody.newBuilder()
+                        .nodeId(nodeInfo.nodeId())
+                        .declineReward(declineReward)
+                        .build()));
+            }
+        }
+        dispatch.stack().commitFullStack();
```

Comment:
Each `dispatchAdmin()` actually commits the dispatch's stack, so this harmless but unnecessary.

Reply:
Ah got it! Will remove in next PR. Thanks @tinker-michaelj